posthog.go

```diff
@@ -314,12 +343,16 @@ func (c *client) GetAllFlags(flagConfig FeatureFlagPayloadNoKey) (map[string]int
 		return nil, err
 	}
 
-	if c.featureFlagsPoller == nil {
-		errorMessage := "specifying a PersonalApiKey is required for using feature flags"
-		c.Errorf(errorMessage)
-		return nil, errors.New(errorMessage)
+	var flagsValue map[string]interface{}
+	var err error
+
+	if c.featureFlagsPoller != nil {
+		flagsValue, err = c.featureFlagsPoller.GetAllFlags(flagConfig)
+	} else {
+		flagsValue, err = c.getAllFeatureFlagsFromDecide(flagConfig.DistinctId, flagConfig.Groups, flagConfig.PersonProperties, flagConfig.GroupProperties)
```

Comment:
didn't add the comments here that the above method has, I'll add them back in for cleanliness